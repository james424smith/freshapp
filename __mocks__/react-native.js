import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
