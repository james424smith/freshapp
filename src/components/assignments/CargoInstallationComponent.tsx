import React from "react";
import { View } from "react-native";
import styles from "../../styles/AssignmentScreenStyles";
import Text from "../StyledText";
import valueOrDefault from "../../common/valueOrDefault";
import { useTheme } from "@react-navigation/native";

type Props = { textLabel: string; textValue: string };

const CargoInstallationComponent = (props: Props) => {
  const { textLabel, textValue } = props;
  const theme = useTheme();
  return (
    <View style={styles.cargoLineStyle}>
      <Text style={{ ...styles.cargoItemLabel, color: theme.colors.text }}>
        {textLabel}:
      </Text>
      <Text style={{ ...styles.cargoItemValue, color: theme.colors.text }}>
        {valueOrDefault(textValue, "-")}
      </Text>
    </View>
  );
};

export default CargoInstallationComponent;
