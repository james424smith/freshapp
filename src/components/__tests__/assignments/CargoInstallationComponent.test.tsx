import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import CargoInstallationComponent from "../../assignments/CargoInstallationComponent";

describe("CargoInstallationComponent snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render CargoInstallationComponent correctly as per the ui made ", () => {
    const { toJSON } = render(
      <CargoInstallationComponent
        textLabel={"Some Label"}
        textValue={"Some Value"}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
