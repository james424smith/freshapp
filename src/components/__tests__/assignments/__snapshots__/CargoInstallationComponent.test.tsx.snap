// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CargoInstallationComponent snapshot test should render CargoInstallationComponent correctly as per the ui made  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#2A2E43",
          "fontFamily": "Roboto-Regular",
        },
        Object {
          "color": "rgb(28, 28, 30)",
          "fontSize": 24,
          "paddingRight": 22.5,
        },
      ]
    }
  >
    Some Label
    :
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#2A2E43",
          "fontFamily": "Roboto-Regular",
        },
        Object {
          "color": "rgb(28, 28, 30)",
          "fontSize": 26.5,
          "fontWeight": "bold",
          "maxWidth": "80%",
        },
      ]
    }
  >
    Some Value
  </Text>
</View>
`;
